documentation/Templates.md

```diff
@@ -67,6 +67,8 @@ Additionally to the [tags and filters](https://stencil.fuller.li/en/latest/built
 | Node | Description |
 | ---- | ----------- |
 | `{% set my_capture_variable %}`<br>`{% endset %}` | node pair allowing you to capture the rendering of nodes inside it, so you can reuse it later by reusing `{{ my_capture_variable }}`.<br>This is useful if you intend to reuse the same content at multiple places — like the `sceneID` is for the `storyboards-default.stencil` template for example. |
+| `{% func my_block a b c %}`<br>`{{args.a}} - {{args.b}} - {{args.c}}`<br>`{% endfunc %}` | node pair allowing you to define a reusable block of nodes inside (like a function), so you can reuse it later.<br>In contrast to the `set` tag, the capturing for `func` is lazy, it will only be processed when actually called. When processed, the block will provide a variable `args` containing all the parameter values.<br>Similar to the `for` tag, this tag provides it's own context scope when called, so variables inside of it are not available outside of it. |
```

Comment:
@djbe I'm a little concerned about adding a lot of complexity for template authors, but I do see a use for it so providing it can be clearly documented and the names and use is very clear I think we can go ahead with it.

---

Just want to add that Jinja2 (which is very similar to Stencil for Python) has similar functionality called "macros" ([docs](http://jinja.pocoo.org/docs/dev/templates/#macros)). Just incase any of the ideas here may be useful.

```html+django
{% macro input(name, value='', type='text', size=20) -%}
    <input type="{{ type }}" name="{{ name }}" value="{{
        value|e }}" size="{{ size }}">
{%- endmacro %}
```

```html+django
<p>{{ input('username') }}</p>
<p>{{ input('password', type='password') }}</p>
```

---

They also have functionality to pass a block down: ([docs](http://jinja.pocoo.org/docs/dev/templates/#call))

```html+django
{% macro render_dialog(title, class='dialog') -%}
    <div class="{{ class }}">
        <h2>{{ title }}</h2>
        <div class="contents">
            {{ caller() }}
        </div>
    </div>
{%- endmacro %}

{% call render_dialog('Hello World') %}
    This is a simple dialog rendered by using a macro and
    a call block.
{% endcall %}
```